### PR TITLE
Patch upgrade Golang from 1.16.4 to 1.16.15

### PR DESCRIPTION
**Pull request by gburton1**
There are a large number of CVEs against prometheus-adapter, and most of them are related to running behind on patches of Golang. This will solve many of them.

Signed-off-by: Greg Burton <9094087+gburton1@users.noreply.github.com>

### CODE REVIEW

**Comment by dgrisonnet** [comment]
Hi @gburton1, do you specifically need this version of Golang or would you be also fine with 1.18.5? I am considering doing a complete dependency update before cutting the new release that would contain that change.

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.9.1
+0.9.2
```
Comment (dgrisonnet) [inline VERSION:1]: this is reserved for when we cut a new release of the project and since this PR is just meant to upgrade the golang version, could you please revert this change?

**Comment by gburton1** [comment]
Any supported Golang major line is fine because they should all offer the same security patches. 1.18.5 works fine!

**Comment by gburton1** [comment]
Reverted the VERSION bump. If you want to handle the jump to 1.18.5 in another PR yourself, since it might involve other changes, I can close this one. But if you expect that to take some time, I would appreciate this trivial patch bump + a new release to get us past a ton of CVEs!

**Comment by dgrisonnet** [comment]
Sure let me do that right now

**Comment by gburton1** [comment]
@dgrisonnet ping me when you have an alternative PR/release. Once we have a new release published with the Golang upgrade, I'll go shop it around to the maintainers of the Prometheus Adapter Helm chart to get a new release there also. This is my path to closing many CVEs in our system against the prom adapter!
